javalib/src/main/scala/java/lang/Character.scala

```diff
@@ -1662,429 +1963,467 @@ object Character {
+      "mathematicalalphanumericsymbols"            -> MATHEMATICAL_ALPHANUMERIC_SYMBOLS,
+      "glagoliticsupplement"                       -> GLAGOLITIC_SUPPLEMENT,
+      "nyiakengpuachuehmong"                       -> NYIAKENG_PUACHUE_HMONG,
+      "wancho"                                     -> WANCHO,
+      "mendekikakui"                               -> MENDE_KIKAKUI,
+      "adlam"                                      -> ADLAM,
+      "indicsiyaqnumbers"                          -> INDIC_SIYAQ_NUMBERS,
+      "ottomansiyaqnumbers"                        -> OTTOMAN_SIYAQ_NUMBERS,
+      "arabicmathematicalalphabeticsymbols"        -> ARABIC_MATHEMATICAL_ALPHABETIC_SYMBOLS,
+      "mahjongtiles"                               -> MAHJONG_TILES,
+      "dominotiles"                                -> DOMINO_TILES,
+      "playingcards"                               -> PLAYING_CARDS,
+      "enclosedalphanumericsupplement"             -> ENCLOSED_ALPHANUMERIC_SUPPLEMENT,
+      "enclosedideographicsupplement"              -> ENCLOSED_IDEOGRAPHIC_SUPPLEMENT,
+      "emoticons"                                  -> EMOTICONS,
+      "ornamentaldingbats"                         -> ORNAMENTAL_DINGBATS,
+      "alchemicalsymbols"                          -> ANATOLIAN_HIEROGLYPHS,
+      "geometricshapesextended"                    -> GEOMETRIC_SHAPES_EXTENDED,
+      "supplementalarrowsc"                        -> SUPPLEMENTAL_ARROWS_C,
+      "chesssymbols"                               -> CHESS_SYMBOLS,
+      "cjkunifiedideographsextensionb"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_B,
+      "cjkunifiedideographsextensionc"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_C,
+      "cjkunifiedideographsextensiond"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_D,
+      "cjkunifiedideographsextensione"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_E,
+      "cjkunifiedideographsextensionf"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_F,
+      "cjkcompatibilityideographssupplement"       -> CJK_COMPATIBILITY_IDEOGRAPHS_SUPPLEMENT,
+      "tags"                                       -> TAGS,
+      "variationselectorssupplement"               -> VARIATION_SELECTORS_SUPPLEMENT,
+      "supplementaryprivateuseareaa"               -> SUPPLEMENTARY_PRIVATE_USE_AREA_A,
+      "supplementaryprivateuseareab"               -> SUPPLEMENTARY_PRIVATE_USE_AREA_B
+    )
+
+    // From: https://unicode.org/Public/UCD/latest/ucd/Blocks.txt
+    // Note:   When comparing block names, casing, whitespace, hyphens,
+    //         and underbars are ignored.
+    //         For example, "Latin Extended-A" and "latin extended a" are equivalent.
+    //         For more information on the comparison of property values,
+    //           see UAX #44: http://www.unicode.org/reports/tr44/
+    private def toNormalizedName(name: String): String = {
+      name.toLowerCase.replaceAll(raw"[\s\-_]", "")
```

Comment:
I believe that the "\s" is incorrect/incomplete here. That is, I am pretty sure that the 
text in the comment is referring to Unicode whitespace (table Zs) while "\s" is referring to the much smaller set of Java whitespace.

As a thought experiment, imagine some person set out to break/test the code by giving an input string which contains an Ogham (Old Irish) space mark (one of my favorite test cases).  I submit that according to the comment, the Ogham character should be removed and that the presented code will not do that.

We are in the land of regex here.  The known & tedious way to loop through each of the input characters and retain it only if it is not in the Unicode Zs table (in scalanative.re2s).   Since replaceAll brings us well into the land of re2s, I could look
to see if re2s has a way to specify * replace Unicode whitespace.

Reply:
At the nanolevel, I think doing the replaceAll first, then the toLowerCase has lower
amortized cost over the expected set of inputs. Cost for text not containing Unicode
whitespace, etc. is probably same (but one always has to look at implementations).
Most names will contain whitespace, etc, so replaceAll first will be a few cycles faster.
Too many hours and exams in Algorithms classes.